src/api/libcellml/importsource.h

```diff
@@ -109,6 +117,161 @@ class LIBCELLML_EXPORT ImportSource: public Entity
      */
     ImportSourcePtr clone() const;
 
+    /**
+     * @brief Add a component to this import source item.
+     *
+     * The added @p component will be treated as an import, and
+     * will be sourced from the combination of url of this import source
+     * item, and the component's own import reference.
+     *
+     * @param component The @c ComponentPtr to add.
+     *
+     * @return @c true if the component was added successfully, @c false
+     * if it already exists in this import.
+     */
+    bool addComponent(const ComponentPtr &component);
+
+    /**
+     * @brief Remove the component at the given @p index.
+     *
+     * Remove the component from this import source at the given @p index.
+     * @p index must be in the range [0, \#component).
+     *
+     * @param index The index of the component to remove.
+     * @param setEmpty Whether or not to set the removed Component's
+     *        importSource to nullptr.  True by default.
```

Comment:
We should something have "`nullptr`" rather than "nullptr".

Reply:
Same elsewhere.

Reply:
I.e. "@c nullptr" rather than "nullptr".

Reply:
Will be addressed in #618